src/main/java/com/treasuredata/client/TDClient.java

```diff
@@ -0,0 +1,527 @@
+        return httpClient.call(request, apiKeyCache);
+    }
+
+    /**
+     * Create a new TDClient that uses the given api key for the authentication.
+     * The new instance of TDClient shares the same HttpClient, so closing this will invalidate the other copy of TDClient instances
+     *
+     * @param newApiKey
+     * @return
+     */
+    @Override
+    public TDClient withApiKey(String newApiKey)
+    {
+        return new TDClient(config, httpClient, Optional.of(newApiKey));
+    }
+
+    @Override
+    public TDClient authenticate(String email, String password)
+    {
+        TDAuthenticationResult authResult = doPost("/v3/user/authenticate", ImmutableMap.of("user", email, "password", password), TDAuthenticationResult.class);
+        return withApiKey(authResult.getApikey());
+    }
+
+    @Override
+    public String serverStatus()
+    {
+        // No API key is requried for server_status
+        return httpClient.call(TDApiRequest.Builder.GET("/v3/system/server_status").build(), Optional.<String>absent());
+    }
+
+    @Override
+    public List<String> listDatabases()
+            throws TDClientException
+    {
+        TDDatabaseList result = doGet("/v3/database/list", TDDatabaseList.class);
+        List<String> tableList = new ArrayList<String>(result.getDatabases().size());
+        for (TDDatabase db : result.getDatabases()) {
+            tableList.add(db.getName());
+        }
+        return tableList;
+    }
+
+    private static Pattern databaseNamePattern = Pattern.compile("^([a-z0-9_]+)$");
+
+    public static String validateDatabaseName(String databaseName)
+    {
+        return validateName(databaseName, "Database");
+    }
+
+    private static String validateTableName(String tableName)
+    {
+        return validateName(tableName, "Table");
+    }
+
+    private static String validateName(String name, String type)
+    {
+        // Validate database name
+        if (name.length() < 3 || name.length() > 256) {
+            throw new TDClientException(
+                    TDClientException.ErrorType.INVALID_INPUT, String.format(
+                    "%s name must be 3 to 256 characters but got %d characters: %s", type, name.length(), name));
+        }
+
+        if (!databaseNamePattern.matcher(name).matches()) {
+            throw new TDClientException(TDClientException.ErrorType.INVALID_INPUT,
+                    String.format("%s name must follow this pattern %s: %s", type, databaseNamePattern.pattern(), name));
+        }
+        return name;
+    }
+
+    @Override
+    public void createDatabase(String databaseName)
+            throws TDClientException
+    {
+        doPost(buildUrl("/v3/database/create", validateDatabaseName(databaseName)));
+    }
+
+    @Override
+    public void createDatabaseIfNotExists(String databaseName)
+            throws TDClientException
+    {
+        if (!existsDatabase(databaseName)) {
+            createDatabase(databaseName);
+        }
+    }
```

Comment:
I dare to say, it's necessary. Since there are already two types of implementation (using existDatabase or try ... catch as you mentioned) to do the same thing. To reduce the support cost, we should provide these functionality. 

Ignoring exception is not ideal because TDHttpClient reports this conflict as an error log, while `existDatabase` approach does not show any error.